contracts/src/libraries/FROST.sol

```diff
@@ -262,54 +262,76 @@ library FROST {
     // ============================================================
 
     /**
-     * @notice Hashes input to a field element for binding factors.
-     * @param input The input bytes.
+     * @notice h1: Hashes input to a field element for calculating binding factors (rho).
+     * @param input The input bytes, which include group public key, message hash, and commitments hash.
      * @return result The field element.
+     * @dev Uses a domain separation tag "FROST-secp256k1-SHA256-v1rho" as required by
+     *      RFC 9591, Section 6.5. Domain separation is critical to prevent an attacker
+     *      from reusing a signature or proof from one context in another. The DST is
+     *      packed into a bytes32 with its length (28 / 0x1c) as the final byte,
+     *      following RFC 9380 recommendations for DST formatting.
```

Comment:
This summary is quite inaccurate IMO:


```suggestion
     * @dev Uses a domain separation tag "FROST-secp256k1-SHA256-v1rho" as required by
     *      RFC 9591, Section 6.5. The DST is
     *      packed into a bytes32 with its length (28 / 0x1c) as the final byte.
```

The "following RFC 9380 recommendations for DST formatting" is not accurate - this is just how we do strings without allocations in Solidity.

Same goes for other hash functions.